packages/http-server-csharp/src/lib/utils.ts

```diff
@@ -587,6 +589,38 @@ export function getModelAttributes(
   return getAttributes(program, entity, cSharpName);
 }
 
+export function getModelExceptionConstructor(
+  program: Program,
+  model: Model,
+  className: string,
+): string {
+  function getDefinedStatusCode() {
+    const statusCodeProperty = new ModelInfo().filterAllProperties(program, model, (p) =>
+      isStatusCode(program, p),
+    );
+
+    if (!statusCodeProperty) return undefined;
+
+    if (statusCodeProperty.type.kind === "Number") {
+      return statusCodeProperty.type.value;
+    }
+
+    return getMinValue(program, statusCodeProperty) ?? undefined;
+  }
+
+  if (!isErrorModel(program, model)) {
+    return "";
+  }
+
+  const definedStatusCode = getDefinedStatusCode();
+  const defaultStatusCode = 400;
+
+  return definedStatusCode
+    ? `public ${className}() : base(${definedStatusCode}) { }`
+    : `public ${className}() : base(${defaultStatusCode}) { }
+       public ${className}(int statusCode) : base(statusCode) { }`;
```

Comment:
I really wonder if we want to have a default constructor if there is no defined status code?

Reply:
it looks kind of unexpected, I removed this, and let the constructor of ` public ${className}(int statusCode) : base(statusCode) { }`

Reply:
I changed this behavior, there is not the option of not passing the statusCode as it is required for Exception , so on that case we are just returning default for that value and passing the reset of the properties